### PR TITLE
Enable parallel queries between tabs a track their responses

### LINKED ISSUE #2077: Keep association between the query response and the tab that made the call

### PR DESCRIPTION
**Pull request by aguillenv**
Resolves #2077

### CODE REVIEW

File: dashboard/src/components/IasqlEditor.tsx [hunk 1]
```diff
@@ -102,11 +101,15 @@ export default function IasqlEditor() {
...
+      exec: () => handleQueryToRunUpdate(selectedDb, editorSelectedTab),
     };
-    editorRef?.current?.editor?.commands?.removeCommand(command);
-    editorRef?.current?.editor?.commands?.addCommand(command);
-  }, [handleQueryToRunUpdate, isRunningSql, selectedDb]);
+    if (editorTabs?.[editorSelectedTab]?.isRunning) {
+      editorRef?.current?.editor?.commands?.removeCommand(command);
+    } else {
+      editorRef?.current?.editor?.commands?.removeCommand(command);
+      editorRef?.current?.editor?.commands?.addCommand(command);
```
Comment (dfellis) [inline dashboard/src/components/IasqlEditor.tsx:110]: How often is this branch executed, do you know? Per keystroke?

File: dashboard/src/components/common/Tab.tsx [hunk 2]
```diff
@@ -14,56 +14,73 @@ export default function Tab({
...
   selectedIndex?: number;
   onChange?: (index: number) => void;
   onTabClose?: (i: number) => void;
+  isLoading?: boolean;
 }) {
+  const getTabContent = (
+    tab: { title: string; action?: () => void; className?: string; width?: string; closable?: boolean },
+    idx: number,
+  ) => {
+    if (tab.closable && selectedIndex === idx && tabs.length > 2) {
```
Comment (dfellis) [inline dashboard/src/components/common/Tab.tsx:31]: Nit: If you moved `selectedIndex` into an argument to this function, could the function be pulled out of the component constructor and used as a normal function? The current structure will recreate a closure every time `Tab` is called.

File: dashboard/src/components/providers/AppProvider.tsx [hunk 3]
```diff
@@ -446,36 +449,31 @@ const middlewareReducer = async (
       let updatedAutoCompleteRes: any;
       let runErr: any;
       try {
```
Comment (dfellis) [inline dashboard/src/components/providers/AppProvider.tsx:451]: Can this outer `try` even catch anything anymore? The only uncaught line is the `dispatch` line, which should be an infallible function call.

I think you can drop the outer try-catch-finally, deleting the catch block and always running the finally block.

File: dashboard/src/components/IasqlEditor.tsx (hunk 1 above)
Comment (aguillenv) [reply dashboard/src/components/IasqlEditor.tsx:110]: No, either when a new db is selected or when something change in the tabs.

**Review by dfellis — APPROVED** [review]

**Review by depombo — APPROVED** [review]

**Review by yrobla — APPROVED** [review]

**Pull request merged**